ci/taos/checker-pr-gateway.sh

```diff
@@ -135,34 +135,70 @@ $checkout_branch
-popd
-echo -e "[DEBUG] Running..."                                         | tee -a $log_file_audit
-echo -e "[DEBUG] Completed."                                         | tee -a $log_file_audit
+
+# @brief checker runner to run the checkers based on the dependency policy
+# Run the first group, then depending on the dependency policy, run the remaining groups
+# @param arguments received by the function
+#  arg1: list of checkers (like format, audit, etc)
+#  arg2: list of checker's name
+#  arg3: list of checker's log file
+#  arg@: all remaining arguments are to be passed to the checkers (common for all the checkers)
+function run_all_checkers(){
+  local -n checker_list=$1
+  local -n name_list=$2
+  local -n logfile_list=$3
+  local checker_args=${@:4}
+
+  local num_checkers=${#checker_list[@]}
+  for (( i=0; i<$num_checkers; i++ )); do
+    local logfile=${logfile_list[$i]}
+
+    cd $dir_ci
+    echo -e "[DEBUG] dir_commit is $dir_commit. This folder is created." | tee    $logfile
+    echo -e "[DEBUG] current path: $(pwd)."                              | tee -a $logfile
+    # Save a log file to debug the cheker
+    echo -e "[DEBUG] ${checker_list[$i]} $checker_args "                 | tee -a $logfile
+    echo -e "[DEBUG] Starting a ${name_list[$i]} checker... "            | tee -a $logfile
+    # Run checker
+    pushd ./taos/
+    ${checker_list[$i]} $checker_args                                    | tee -a $logfile &
+    local pid=$!
+    popd
+    echo -e "[DEBUG] Running..."                                         | tee -a $logfile
+
+    # if the dependency between groups is to be enforced and current group fails, skip remaining groups
+    if [[ $dependency_policy_between_groups == 1 ]]; then
+      wait $pid
+      if [[ $? != 0 ]]; then
+        break
+      fi
+    fi
+  done
+}
+
+# --------------------------- Create and run checkers -------------------------------------
+# Create an empty list for the checkers to be run with the corresponding elements
+# The order of the checkers in the list creates a dependency between them determining the order in which they will run
+# 1) Checker itself to be run
+# 2) Name of the checker
+# 3) Log file to log the output of the checker
+checker_list=()
+checker_name_list=()
+checker_logfile_list=()
+
+# Ready a checker: format
+checker_list+=("./checker-pr-format-async.sh")
+checker_name_list+=("format")
+checker_logfile_list+=("${dir_ci}/${dir_commit}/checker-pr-format.log")
+echo -e "[DEBUG] Added ${checker_name_list[-1]} checker"
+
+# Ready a checker: audit
+checker_list+=("./checker-pr-audit-async.sh")
+checker_name_list+=("audit")
+checker_logfile_list+=("${dir_ci}/${dir_commit}/checker-pr-audit.log")
+echo -e "[DEBUG] Added ${checker_name_list[-1]} checker"
+
+# Run all the checkers
+echo -e "[DEBUG] Running all checkers"
+run_all_checkers checker_list checker_name_list checker_logfile_list $1 $2 $3 $4 $5 $6
```

Comment:
For readability and maintenance, 
`s/checker_list/checker_cmd_list/g `

Reply:
Done.